### ISSUE TITLE
Vert.x MQTT Ver5.0 support

### ISSUE THREAD

**Issue by MrXionGe** [vert-x3/vertx-mqtt#222]
#### module: Vert.x MQTT
Do we have plans to implement full support for MQTT5.0?
I saw that it was mentioned in the documentation, MQTT5 support is about 90+% complete.
I want to create an MQTT Broker using "vert. x MQTT".

**Comment by HerrDerb** [comment]
Hi @vietj. Is there some developing on the way that soon will be available for early adopters or is this still  in the drawer?

**Comment by vietj** [comment]
we are still missing a few features @HerrDerb currently no contributor is implementing it

**Comment by HerrDerb** [comment]
@vietj Is there an overview of what is missing? I might be interested to contribute.

**Comment by vietj** [comment]
To be honest I do not have a status on this, if you can evaluate what is missing and are willing to contribute then we can work accomplishing this goal

**Comment by HerrDerb** [comment]
@vietj when implementing mqtt5 for the client, would you preferer a unified client or two separate classes?

**Comment by vietj** [comment]
Actually I have no strong opinion about it, perhaps you can make some research and we could schedule a call to discuss all the options for supporting MQTT v5 ?

**Comment by HerrDerb** [comment]
Yes, I can try. This might take a moment, as currently, I am doing this on the side. I'll come back to you.

**Comment by vietj** [comment]
@HerrDerb sure, take your time

**Comment by alamothe** [comment]
Please give us MQTT 5 client support.

**Comment by dometec** [comment]
Hi! I'm implementing MQTT v5 on the client side, see #265. I agree with @HerrDerb — there are already many open-source brokers available, including Java-based ones such as Apache Artemis, so maintaining another one here would likely be a waste of time.

It might be useful to split the repository into separate client and server projects. That way, anyone interested in improving the server side would be free to do so independently.

Moreover, the server component could potentially evolve into a standalone server, and having a dedicated repository for it would make the overall structure clearer and less confusing.

**Comment by vietj** [comment]
thanks @dometec I will follow this effort

**Comment by ppatierno** [comment]
The goal of the MQTT server part was not about developing a full MQTT broker, but just a server facing component which is a totally different thing. With just the MQTT server part able to handle clients connections and MQTT protocol semantic, you can use it to develop a full broker which will define its way to store messages (i.e. a file, a db, or something else) or even to develop a bridge. I don't like the idea that the Vert.x MQTT is going to have client on 5.0 and server on 3.1.1.
I think the proposal should be to provide 5.0 on both.

**Comment by vietj** [comment]
@ppatierno there is already MQTT v5 server support @ppatierno it is not 100% complete, so we should evaluate what is missing

**Comment by vietj** [comment]
> The goal of the MQTT server part was not about developing a full MQTT broker, but just a server facing component which is a totally different thing. With just the MQTT server part able to handle clients connections and MQTT protocol semantic, you can use it to develop a full broker which will define its way to store messages (i.e. a file, a db, or something else) or even to develop a bridge. I don't like the idea that the Vert.x MQTT is going to have client on 5.0 and server on 3.1.1. I think the proposal should be to provide 5.0 on both.

I agree with you, it can be useful in many cases

**Comment by ppatierno** [comment]
> @ppatierno there is already MQTT v5 server support @ppatierno it is not 100% complete, so we should evaluate what is missing

@vietj You are absolutely right! I forgot about that. Yeah we should try to track what's missing I will set it on my TODO list, at least for opening issues to track them. But no ETA for that.

**Comment by ZHANGSHUNLIN** [comment]
I've finally seen the updates for the mqtt5 client and look forward to the official release.

**Comment by HerrDerb** [comment]
@ppatierno any timeline for a first mqtt 5 client release? 😬
